### PR TITLE
Add reservation views

### PR DESCRIPTION
**Pull request by MStodolkiewicz**
Added views for reservations

Added methods to reservation controller and routes

### CODE REVIEW

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -4,14 +4,28 @@ class ReservationsController < ApplicationController
   before_action :set_reservation, only: %i[show edit update destroy]
   before_action :authenticate_user!
 
+  def find_reservations_by_user
```
Comment (w0st) [inline app/controllers/reservations_controller.rb:7]: It looks like you use [rubocop](https://github.com/MStodolkiewicz/Monte_Cinema/blob/main/.rubocop.yml), but the syntax violates the defined rules by you.

Do you need help with configuring rubocop plugin for your IDE?

```
Offenses:

app/controllers/reservations_controller.rb:8:47: C: [Correctable] Layout/TrailingWhitespace: Trailing whitespace detected.
    @reservations = Reservation.where(user_id: 
                                              ^
app/controllers/reservations_controller.rb:10:5: C: [Correctable] Layout/MultilineMethodCallIndentation: Align .joins with .where on line 8.
    .joins(:seance).includes([:user, seance:[:movie]])
    ^^^^^^
app/controllers/reservations_controller.rb:10:38: C: [Correctable] Style/HashAsLastArrayItem: Wrap hash in { and }.
    .joins(:seance).includes([:user, seance:[:movie]])
                                     ^^^^^^^^^^^^^^^
app/controllers/reservations_controller.rb:10:44: C: [Correctable] Layout/SpaceAfterColon: Space missing after colon.
    .joins(:seance).includes([:user, seance:[:movie]])
```

File: app/controllers/reservations_controller.rb [hunk 2]
```diff
@@ -4,14 +4,28 @@ class ReservationsController < ApplicationController
...
+    @reservations = Reservation.where(user_id: 
+      User.find_by(email: params[:email]))
+    .joins(:seance).includes([:user, seance:[:movie]])
```
Comment (w0st) [inline app/controllers/reservations_controller.rb]:
```suggestion
    .includes([:user, seance:[:movie]])
```

Do you need `joins` here? :thinking:

File: app/controllers/reservations_controller.rb [hunk 3]
```diff
@@ -4,14 +4,28 @@ class ReservationsController < ApplicationController
   before_action :set_reservation, only: %i[show edit update destroy]
   before_action :authenticate_user!
 
+  def find_reservations_by_user
+    @reservations = Reservation.where(user_id: 
+      User.find_by(email: params[:email]))
```
Comment (w0st) [inline app/controllers/reservations_controller.rb]: This line is an ideal candidate for another method :wink:

File: app/views/movies/show.html.erb [hunk 4]
```diff
@@ -19,7 +19,12 @@
   <div class="border border-dark" style="float:left; margin-right: 2%; margin-bottom:2%; width: 15%;">
       <%= render seance %>
 
+      <%= link_to "Make reservation", new_reservation_path(:seance_id => seance.id), class: "btn btn-outline-success" %>
```
Comment (w0st) [inline app/views/movies/show.html.erb]: You can use new Ruby hash syntax :wink: 
```suggestion
      <%= link_to "Make reservation", new_reservation_path(seance_id: seance.id), class: "btn btn-outline-success" %> 
```

File: app/views/reservations/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    </div>
+  <% end %>
+
+  <div>
+    <%= form.label :seance_id, style: "display: block" %>
+    <%= form.text_field :seance_id %>
+  </div>
+
+  <div>
+    <%= form.label :user_id, style: "display: block" %>
```
Comment (w0st) [inline app/views/reservations/_form.html.erb:20]: Is it intended to allow the user to make a reservation in the name of another user? :thinking: 
Looking at the implementation of `ReservationsController#create` it looks like I can create a reservation for any user in the system.

File: config/routes.rb [hunk 6]
```diff
@@ -5,8 +5,11 @@
     resources :seances, except: [:show, :index] do
       resources :reservations
     end
-  end     
-resources :seances, :reservations
+  end 
+  resources :seances
+  resources :reservations
+  get '/find_reservations_by_user', to: "reservations#find_reservations_by_user"
```
Comment (w0st) [inline config/routes.rb]: You can consider [collection routes ](https://guides.rubyonrails.org/routing.html#adding-collection-routes)for this purpose :wink:

**Review by w0st — COMMENTED** [review]
Consider my comments, please.

File: app/controllers/reservations_controller.rb (hunk 3 above)
Comment (MStodolkiewicz) [reply app/controllers/reservations_controller.rb]: So should make private method, let's say `get_user_by_email` and simply call it here right?

File: app/controllers/reservations_controller.rb (hunk 2 above)
Comment (MStodolkiewicz) [reply app/controllers/reservations_controller.rb]: Without `joins` I'm not able to order records by `start_time`. It doesn't see this column

File: app/controllers/reservations_controller.rb (hunk 1 above)
Comment (MStodolkiewicz) [reply app/controllers/reservations_controller.rb:7]: No, sorry, I had to stash this for a second and I probably run rubocop before popping it.
Rubocop returned offenses now so it's fixed :).

File: config/routes.rb (hunk 6 above)
Comment (MStodolkiewicz) [reply config/routes.rb]: Closed this routes in collection, thanks :D

File: app/views/reservations/_form.html.erb (hunk 5 above)
Comment (MStodolkiewicz) [reply app/views/reservations/_form.html.erb:20]: Sorry, I should have completely delete this method before making PR, it was just for testing and it's gonna look completely different when I'll implement tickets

File: app/controllers/reservations_controller.rb (hunk 2 above)
Comment (w0st) [reply app/controllers/reservations_controller.rb]: valid point :+1:

File: app/controllers/reservations_controller.rb (hunk 3 above)
Comment (w0st) [reply app/controllers/reservations_controller.rb]: Yes, it should increase readability.

Comment (MStodolkiewicz) [reply app/controllers/reservations_controller.rb]: Okey, did that, thank you for all the comments! :D

**Pull request merged**